carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,260 @@
+            if a download is required.
+        models_home : string, optional
+            The directory in which to cache data; see :func:`get_models_home`.
+        kws : keys and values, optional
+            Additional keyword arguments are passed to passed through to the read model function
+        data : data.api.Data Class
+            Correct dataset for ML model
+        ext : String
+            File extension of saved ML model file
+        """
+        # check if dataset fits to ML model
+        assert data.name == data_name
+
+        self._continuous = data.continous
+        self._categoricals = data.categoricals
+
+        self._data_name = data_name
+        self._model = load_model(model_type, data_name, ext, cache, models_home, **kws)
+
+        if ext == "pt":
+            self._backend = "pytorch"
+        elif ext == "h5":
+            self._backend = "tensorflow"
+        else:
+            raise Exception("Model type not in catalog")
+
+        self._name = model_type + "_" + data_name
+
+        # Preparing pipeline components
+        self._scaler = preprocessing.MinMaxScaler().fit(data.raw[self._continuous])
+
+        if data_name == "adult":
+            self._feature_input_order = [
+                "age",
+                "fnlwgt",
+                "education-num",
+                "capital-gain",
+                "capital-loss",
+                "hours-per-week",
+                "workclass_Private",
+                "marital-status_Non-Married",
+                "occupation_Other",
+                "relationship_Non-Husband",
+                "race_White",
+                "sex_Male",
+                "native-country_US",
+            ]
+            self._encodings = (
+                [  # Encodings should be built in the get_dummy way: {column}_{value}
+                    "workclass_Private",
+                    "marital-status_Non-Married",
+                    "occupation_Other",
+                    "relationship_Non-Husband",
+                    "race_White",
+                    "sex_Male",
+                    "native-country_US",
+                ]
+            )
+        else:
+            raise Exception("Model for dataset not in catalog")
+
+    def pipeline(self, df):
+        """
+        Transforms input for prediction into correct form.
+        Only possible for DataFrames without preprocessing steps.
+
+        Recommended to use to keep correct encodings, normalization and input order
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            Contains unnormalized and
+
+        Returns
+        -------
+        output : pd.DataFrame
+            Prediction input in correct order, normalized and encoded
+
+        """
+        output = df.copy()
+
+        # Normalization
+        output[self._continuous] = self._scaler.transform(output[self._continuous])
+
+        # Encoding
+        output[self._encodings] = 0
```

Comment:
Its not very clear what you doing here and the following lines. Also It seems like functionality that could be grouped in a seperate function, where you could use the docstring to explain

Reply:
The idea was to encode arbitrary input instances with respect to the encoded feature names, to keep it consistent.
For example, using pandas get_dummies without context to the whole dataset could lead to inconsistent encodings for different inputs.

In future I would like to find a different solution, such that enconding mechanism in the data catalog is consistent to the one in the MLModel pipeline.

Reply:
As discusssed, this logic would go away in a follow up PR, so happy to leave it like this for now